eng/scripts/Language-Settings.ps1

```diff
@@ -6,28 +6,65 @@ $packagePattern = "*.nupkg"
 $MetadataUri = "https://raw.githubusercontent.com/Azure/azure-sdk/master/_data/releases/latest/dotnet-packages.csv"
 $BlobStorageUrl = "https://azuresdkdocs.blob.core.windows.net/%24web?restype=container&comp=list&prefix=dotnet%2F&delimiter=%2F"
 
-function Get-dotnet-PackageInfoFromRepo ($pkgPath, $serviceDirectory, $pkgName)
+function Get-dotnet-PackageInfoFromRepo
 {
-  $projectPath = Join-Path $pkgPath "src" "$pkgName.csproj"
-  if (Test-Path $projectPath)
+  [CmdletBinding()]
+  Param(
+    [Parameter(Position = 0)]
+    [string]$pkgDirectoryPath,
+    [Parameter(Position = 1)]
+    [string]$serviceDirectoryName,
+    $pkgPath,
+    $serviceDirectory,
+    $pkgName
+  )
+
+  if ($pkgName)
   {
-    $projectData = New-Object -TypeName XML
-    $projectData.load($projectPath)
-    $pkgVersion = Select-XML -Xml $projectData -XPath '/Project/PropertyGroup/Version'
-    $sdkType = "client"
-    if ($pkgName -match "\.ResourceManager\." -or $pkgName -match "\.Management\.")
+    $projectPath = Join-Path $pkgPath "src" "$pkgName.csproj"
+    if (Test-Path $projectPath)
     {
-      $sdkType = "mgmt"
+      $projectData = New-Object -TypeName XML
+      $projectData.load($projectPath)
+      $pkgVersion = Select-XML -Xml $projectData -XPath '/Project/PropertyGroup/Version'
+      $sdkType = "client"
+      if ($pkgName -match "\.ResourceManager\." -or $pkgName -match "\.Management\.")
+      {
+        $sdkType = "mgmt"
+      }
+      $pkgProp = [PackageProps]::new($pkgName, $pkgVersion, $pkgPath, $serviceDirectory)
+      $pkgProp.SdkType = $sdkType
+      $pkgProp.IsNewSdk = $pkgName.StartsWith("Azure")
+      $pkgProp.ArtifactName = $pkgName
+      return $pkgProp
     }
-    $pkgProp = [PackageProps]::new($pkgName, $pkgVersion, $pkgPath, $serviceDirectory)
-    $pkgProp.SdkType = $sdkType
-    $pkgProp.IsNewSdk = $pkgName.StartsWith("Azure")
-    $pkgProp.ArtifactName = $pkgName
-    return $pkgProp
   }
-  else
+  else 
   {
-    return $null
+    $pkgProperties = @()
+    if (Test-Path (Join-Path $pkgDirectoryPath "src"))
+    {
+      $projectFilePaths = Get-ChildItem (Join-Path $pkgDirectoryPath "src") "*.csproj" -File
```

Comment:
I think it is save to assert that there is only one csproj here and if we find more then one we should error/warn and only take the first.

Reply:
Once we do that we should be able to reuse most of this function.

Reply:
Again am using different argument names inthe different code blocks. Also what if the first .csproj is the wrong one?

Reply:
I think the assertion here is that if we have two ```*.csproj``` files in the one directory here that it is a failure condition. That shouldn't happen.